src/credentials/attestation_verifier/DeviceAttestationVerifier.cpp

```diff
@@ -111,5 +113,42 @@ void SetDeviceAttestationVerifier(DeviceAttestationVerifier * verifier)
     gDacVerifier = verifier;
 }
 
+ByteSpan DupByteSpanHelper(const ByteSpan & span_to_dup)
+{
+    uint8_t * bytes = (uint8_t *) chip::Platform::MemoryAlloc(span_to_dup.size());
+    memcpy(bytes, span_to_dup.data(), span_to_dup.size());
+    return ByteSpan(bytes, span_to_dup.size());
+}
+
+DeviceAttestationVerifier::AttestationDeviceInfo::AttestationDeviceInfo(const AttestationInfo & attestationInfo) :
+    mPaiDerBuffer(DupByteSpanHelper(attestationInfo.paiDerBuffer)), mDacDerBuffer(DupByteSpanHelper(attestationInfo.dacDerBuffer))
+{
+    mAttestationElementsBufferCopy = chip::Platform::MemoryAlloc(attestationInfo.attestationElementsBuffer.size());
```

Comment:
Seems like we don't really need to save this if the CD is the only thing we need to get of this? Perhaps just allocate space for the extracted CD?

Reply:
Okay can certainly do that instead.